### PR TITLE
HRJS-61 HRJS-62

### PR DESCRIPTION
**Pull request by galderz**
https://issues.jboss.org/browse/HRJS-61
https://issues.jboss.org/browse/HRJS-62

### CODE REVIEW

File: memory-profiling/infinispan_memory_many_get.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        console.log("After first put, heap used: " + heapUseAfterPut);
+        console.log("After many gets, heap used: " + heapAfterManyGets);
+        console.log("After final get, heap used: " + heapAfterFinalGet);
+
+        return client.disconnect();
+      });
+    });
+  });
+}).catch(function(err) {
+  console.log("Error: " + err);
```
Comment (gustavonalle) [inline memory-profiling/infinispan_memory_many_get.js]: I got an error when running this file. After 5s or so:
```
...
[get] Value:  test
[get] Value:  test
Error: TypeError: global.gc is not a function
````

File: memory-profiling/infinispan_memory_one_get.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    heapdump.writeSnapshot('/tmp/' + Date.now() + '.heapsnapshot');
+  });
+
+  return dumpAfterGet2.then(function test_disconnect() {
+    return client.disconnect();
+  });
+
+}).catch(function(err) {
+  console.log("connect error", err);
```
Comment (gustavonalle) [inline memory-profiling/infinispan_memory_one_get.js]: This one fails faster:
```
[2018-05-22 16:27:06.449] [DEBUG] io_1_conn_2 - Connecting to 127.0.0.1:11222
[2018-05-22 16:27:06.454] [DEBUG] io_1_conn_2 - Connected to 127.0.0.1:11222
[2018-05-22 16:27:06.455] [DEBUG] io_1 - New router installed: FixedRouter(conn=127.0.0.1:11222@conn_2)
[2018-05-22 16:27:06.455] [DEBUG] client - Invoke ping(msgId=3)
[2018-05-22 16:27:06.455] [TRACE] encoder - Encode operation with topology id 0
[2018-05-22 16:27:06.457] [TRACE] io_1 - Write buffer(msgId=3) to 127.0.0.1:11222@conn_2: A00319170A6E616D65644361636865000300
[2018-05-22 16:27:06.460] [TRACE] decoder - Read header(msgId=3): opCode=24, status=0, hasNewTopology=0
[2018-05-22 16:27:06.460] [TRACE] decoder - Call decode for request(msgId=3)
[2018-05-22 16:27:06.460] [TRACE] io_1_conn_2 - Complete success for request(msgId=3) with undefined
[2018-05-22 16:27:06.462] [TRACE] io_1_conn_2 - After decoding request(msgId=3), buffer size is 5, and offset 5
[2018-05-22 16:27:06.462] [DEBUG] client - Started Infinispan Client(id=io_1,router=FixedRouter(conn=127.0.0.1:11222@conn_2))
[2018-05-22 16:27:06.462] [DEBUG] client - Invoke put(msgId=4,key=memory-one-get,value=test,opts=undefined)
[2018-05-22 16:27:06.463] [TRACE] encoder - Encode operation with topology id 0
[2018-05-22 16:27:06.463] [TRACE] io_1 - Write buffer(msgId=4) to 127.0.0.1:11222@conn_2: A00419010A6E616D656443616368650003000E6D656D6F72792D6F6E652D676574770474657374
[2018-05-22 16:27:06.466] [TRACE] decoder - Read header(msgId=4): opCode=2, status=0, hasNewTopology=0
[2018-05-22 16:27:06.466] [TRACE] decoder - Call decode for request(msgId=4)
[2018-05-22 16:27:06.466] [TRACE] io_1_conn_2 - Complete success for request(msgId=4) with undefined
[2018-05-22 16:27:06.467] [TRACE] io_1_conn_2 - After decoding request(msgId=4), buffer size is 5, and offset 5
[2018-05-22 16:27:06.571] [DEBUG] client - Invoke get(msgId=5,key=memory-one-get)
[2018-05-22 16:27:06.571] [TRACE] encoder - Encode operation with topology id 0
[2018-05-22 16:27:06.572] [TRACE] io_1 - Write buffer(msgId=5) to 127.0.0.1:11222@conn_2: A00519030A6E616D656443616368650003000E6D656D6F72792D6F6E652D676574
[2018-05-22 16:27:06.572] [TRACE] decoder - Read header(msgId=5): opCode=4, status=0, hasNewTopology=0
[2018-05-22 16:27:06.573] [TRACE] decoder - Call decode for request(msgId=5)
[2018-05-22 16:27:06.573] [TRACE] io_1_conn_2 - Complete success for request(msgId=5) with test
[2018-05-22 16:27:06.573] [TRACE] io_1_conn_2 - After decoding request(msgId=5), buffer size is 10, and offset 10
[get1] value: test
connect error TypeError: global.gc is not a function
    at test_dump_after_get1 (/home/gfernandes/github/js-client/memory-profiling/infinispan_memory_one_get.js:28:12)
    at tryCallOne (/home/gfernandes/github/js-client/node_modules/promise/lib/core.js:37:12)
    at /home/gfernandes/github/js-client/node_modules/promise/lib/core.js:123:15
    at flush (/home/gfernandes/github/js-client/node_modules/asap/raw.js:50:29)
    at _combinedTickCallback (internal/process/next_tick.js:73:7)
    at process._tickCallback (internal/process/next_tick.js:104:9)
```

File: memory-profiling/infinispan_memory_many_get.js (hunk 1 above)
Comment (gustavonalle) [reply memory-profiling/infinispan_memory_many_get.js]: Do I need to pass in some flags to node to enable ```gc``` ?

**Comment by galderz** [comment]
Expired certificate problems are being tracked by [HRJS-53](https://issues.jboss.org/browse/HRJS-53). I'll work on that next, once this higher prio issue has been fixed ;)

File: memory-profiling/infinispan_memory_many_get.js (hunk 1 above)
Comment (galderz) [reply memory-profiling/infinispan_memory_many_get.js]: Yeah, requires a special flag. I've pushed a commit to add documentation around this.

File: memory-profiling/infinispan_memory_one_get.js (hunk 2 above)
Comment (galderz) [reply memory-profiling/infinispan_memory_one_get.js]: Same thing

**Pull request merged**